core/constants_submission.py

```diff
@@ -0,0 +1,46 @@
+from core.constants import SIGNIFICANT_BINARY_LABEL_COLUMNS, SIGNIFICANT_RIDGE_COLUMNS
+from core.model_induction import train_classifier_tree, train_decision_tree
+from core.model_induction_nn import train_classifier_network
+from core.model_regression import train_linear_regression
+from core.modifiers import modifier_filter_columns, modify_model, modifier_balance_binary_data
+
+SUBMISSION1_RIDGE_FEATURE_SET_COUNTS = [1, 3, 7, 5, 10]
+SUBMISSION1_PROPAGATION_FEATURE_SET_COUNTS = [3, 5, 10, 15, 20]
+
+# Model sets to use for submission 2
+SUBMISSION2_MODEL_SETS = [
+    (
+        'All feature regression test (Jonathan 1)',
+        train_decision_tree, [],
+        train_linear_regression, []
+    ),
+    (
+        '10-feature regression test (Jonathan 2)',
+        train_decision_tree, [],
+        train_linear_regression, [
+            modifier_filter_columns(SIGNIFICANT_RIDGE_COLUMNS[:10]),
+        ]
+    ),
+    (
+        'Respective top 3',
+        train_classifier_tree, [
+            modifier_filter_columns(SIGNIFICANT_BINARY_LABEL_COLUMNS[:3]),
+        ], train_linear_regression, [
+            modifier_filter_columns(SIGNIFICANT_RIDGE_COLUMNS[:3]),
+        ]
+    ),
+    (
+        'Complex model setup example',
+        modify_model(train_classifier_tree, max_depth=10), [
+            modifier_filter_columns(SIGNIFICANT_BINARY_LABEL_COLUMNS[:10]),
+            modifier_balance_binary_data(skew_false=8)
+        ], train_linear_regression, [
+            modifier_filter_columns(SIGNIFICANT_RIDGE_COLUMNS[:10]),
+        ]
+    ),
+    (
+        'Neural network classifier',
+        modify_model(train_classifier_network, epochs=50, batch_size=100), [],
+        train_linear_regression, []
+    ),
+]
```

Comment:
We can use this to set up complex models for our submissions.

I included the two that I want for submission 2.

The rest are examples on how this can be done.

1. Initially a name is provided for each model set entry.

2. First the training function for the induction model is specified, if we want to add custom `kwargs` we can use the `modify_model(train_model, **kwargs)` function to make the easier so we don't have to use an anonymous function.

3. Next a list of modifiers is passed for the induction model, allowing the data to be processed sequentially through the modifiers before being used for training.

4. The previous two steps (2, 3) are then used to specify the regression functionality.

This is also explained in the `_predict_model_sets(...)` function code comments, which I will tag.

Reply:
Intuitively might be a bit more self-descriptive to use a `ModelSet` dataclass for these so we can use kwargs and require fields but tuples are fine too

Reply:
Intuitively might be a bit more self-descriptive to use a `ModelSet` dataclass for these so we can use kwargs and require fields but tuples are fine too

Reply:
Implemented this, much nicer to declare the model sets now.